addons/website_slides/models/slide_slide.py

```diff
@@ -240,12 +241,16 @@ class Slide(models.Model):
         ('exclusion_html_content_and_url', "CHECK(html_content IS NULL OR url IS NULL)", "A slide is either filled with a url or HTML content. Not both.")
     ]
 
-    @api.depends('slide_category', 'source_type', 'image_binary_content')
+    @api.depends('slide_category', 'source_type', 'image_binary_content', 'document_binary_content')
     def _compute_image_1920(self):
         for slide in self:
             if slide.slide_category == 'infographic' and slide.source_type == 'local_file' and slide.image_binary_content:
                 slide.image_1920 = slide.image_binary_content
-            elif not slide.image_1920:
+            elif slide.slide_category == 'document' and slide.slide_type == 'pdf' and slide.document_binary_content:
+                pdf_preview = pdf_utils.render_pdf_first_page_as_image(slide.document_binary_content)
+                if pdf_preview:  # Only update if preview has succeeded because preview can also be sent by the front
+                    slide.image_1920 = pdf_preview
+            if not slide.image_1920:
```

Comment:
This part is normally not necessary anymore, since computed fields have now a default value like other fields...